addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -847,6 +860,39 @@ options.Class.include({
             'reset_view_arch': resetViewArch,
         });
     },
+    /**
+     * Caches the rpc.
+     *
+     * @override
+     */
+    async _rpcRequestCached() {
+        const cacheId = JSON.stringify(...arguments);
+        if (!this._rpcCache[cacheId]) {
+            this._rpcCache[cacheId] = this.rpc(...arguments);
+        }
+        return this._rpcCache[cacheId];
+    },
+
+    /**
+     * @override
+     */
+    async _computeWidgetVisibility(widgetName, params) {
+        const _super = this._super.bind(this);
+        if (
+            [
+                "footer_language_selector_label_opt",
+                "footer_language_selector_opt",
+                "header_language_selector_label_opt",
+                "header_language_selector_opt",
+            ].includes(widgetName)
+        ) {
+            this._languages = await this._rpcRequestCached("/website/get_languages");
+            if (this._languages.length === 1) {
+                return false;
+            }
+        }
+        return _super(...arguments);
+    }, 
```

Comment:
If I am not mistaken, you are in the base website definition of all website snippet options here. You should in fact only impact the options related to the language selection. Probably one for the footer and one for the header actually?

Reply:
Done and indeed it's better this way.